Documentation/edit bootstrap

wabernat opened the pull request:
**Component**: documentation

**Context**: 

**Summary**: Deep dive on Bootstrap instructions. Simplified some steps, reordered chapter for logical sequencing. 

**Acceptance criteria**: Like it? 

---

See: #ZENKO-2421

alexandre-allard committed:
examples: rename example operator

Rename the operator to match the pattern
<solution-name>-operator:
example-operator -> example-solution-operator

alexandre-allard committed:
examples: add a --config flag to operator

Add an option to pass a configuration to the operator
For the moment, only the configuration for the registry
prefixes are handled.

wabernat committed:
Deep edit on introduction.

Tightened language throughout.

Removed redundant :term: fields (one per term per page)

Corrected incorrect code labels.

etc.

wabernat committed:
Walked the steps on the bootstrap process. Made changes to conform to my naive
user eperience. Simplified some of the commands. Gave (I hope) correct
orthography on root (#) vs. user ($) command prompts.

Reordered chapter (index.rst) to make the procedure make more sense: Users
cannot access GUI until PVs are allocated for Prometheus and Alertmanager, so
PV activation must come *before* GUI instructions.

Trivial change to introduction (for euphony).

wabernat committed:
Walked the steps on the bootstrap process. Made changes to conform to my naive
user eperience. Simplified some of the commands. Gave (I hope) correct
orthography on root (#) vs. user ($) command prompts.

Reordered chapter (index.rst) to make the procedure make more sense: Users
cannot access GUI until PVs are allocated for Prometheus and Alertmanager, so
PV activation must come *before* GUI instructions.

Trivial change to introduction (for euphony).

bert-e commented:
# Hello wabernatscality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




There is a conflict between your branch `documentation/edit_bootstrap` and the
destination branch `development/2.4`.

Please resolve the conflict on **the feature branch** (`documentation/edit_bootstrap`).

```bash
 $ git fetch
 $ git checkout origin/documentation/edit_bootstrap
 $ git merge origin/development/2.4
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:documentation/edit_bootstrap
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `documentation/edit_bootstrap` into
`development/2.4`
* [w/2.5/documentation/edit_bootstrap](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/documentation/edit_bootstrap) will be merged into `development/2.5` 
* [w/2.6/documentation/edit_bootstrap](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/documentation/edit_bootstrap) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

alexandre-allard committed:
build: allow the configuration of the docker socket

Allow to change the socket used to communicate with
docker for skopeo during the build of the example
solution
We need it, because we use a unix socket when building
locally, but a tcp one when in the CI.

Refs: #1605

alexandre-allard committed:
ci: add a step to build example-solution

This step will be used to provide an example
solution artifact to be able to tests the
whole Solution framework.

Refs: #1605

alexandre-allard committed:
ci: enable building of example solution in post-merge

First step to be able to tests the Solution framework

Refs: #1605

alexandre-allard committed:
ci: add a pod builder definition for example-solution

Create a docker image with all necessary dependencies
to build example-solution in the CI

Refs: #1605

wabernat committed:
Merge branch 'development/2.4' of http://github.com/Scality/metalk8s into development/2.4

Ebaneck committed:
script: make use of kubeconfig in solutions script

While using the solutions script, users fall into the following issues
The connection to the server localhost:8080 was refused - did you specify the right host or port?`

This is as a result of no kubeconfig flag specified while perform the kubectl commands.
This addition is especially useful since we need to automate example-solutions in the CI

TeddyAndrieux committed:
eve: Generate simple junit for each eve steps

Currently we do not have any report for build status. This commit add a
very very simple script to generate really simple junit for each steps
of the build. 1 junit with 1 test suite and 1 test case per step.
A better junit reporting should be done directly by eve in the futur,
this junit just allow us to have a report giving use the result of each
build steps

Fixes: #2267

TeddyAndrieux committed:
eve: HaltonFailure on lint step

Before this PR, lint step do not halt on failure, meaning that if lint
step we continue with next steps of this stage, which setting final
status to success, so if lint fail we still set final status of the lint
step to successfull, which is wrong

slaperche-scality committed:
examples: no longer use a singleton for the configuration

Instead of using a singleton pattern for the Operator
configuration, we now pass the configuration through
the manager to make it accessible in controllers.

Signed-off-by: Sylvain Laperche <sylvain.laperche@scality.com>

thomasdanan committed:
basic alerting requirements:

draft version
add user story for silencing an alert
highlight the need of having persistent custos
fix spelling issues
add link to am configuration doc
fix formatting issue

alexandre-allard committed:
examples: fail if container image is not found in cfg

TeddyAndrieux committed:
eve: Git pull current branch for buildprev step

To build final status and junit file we need to have access to the git
repo, so first pull current branch to build junit file before building
the previous version

Sees: #2267

alexandre-allard committed:
salt: change operator config format

In the operator configuration, we know list the
repositories endpoints & images by solution version.
We need this to tell the Operator where to fetch
the images for a specific solution version.

alexandre-allard committed:
examples: add a default config for example-solution

This default configuration allow action on apiGroups
example-solution.metalk8s.scality.com for the Operator
to be able to monitor CRs

alexandre-allard committed:
examples: add makefile rule to build example solution config

alexandre-allard committed:
salt: mount solution directory in salt master

this change will ease the management and deployment
of solutions as the salt-master is able to read
the resources from the solutions archives

Refs: #2260

alexandre-allard committed:
salt: use metalk8s kubernetes state instead of renderer

We use the metalk8s_kubernetes state instead of the
kubernetes renderer, otherwise it may fails while
parsing the CRDs yaml.
It also makes what is done clearer and simpler
than before.

Refs: #2260

Ebaneck committed:
tests: Add tests for Solutions CLI

This commit adds tests to validate the Solutions CLI

Formally, we added eve steps to automate the deployment and testing of
Solutions CLI.

In an attempt to expand our test-suite, we have rewritten the test-suite
in Python bdd

alexandre-allard committed:
scripts: reconfigure salt master when importing Solution

The Solution archives are now mounted inside the
Salt master container to ease the access to the
solutions resource definitions (e.g. CRDs)

Refs: #2260

alexandre-allard committed:
build: add yum-utils to redhat downloaded packages

download yum-utils RedHat package in the buildchain.
yum-utils is used to provide repoquery to be able
to lookup for packages and their dependencies and
enforce certain versions.

Refs: #2334

alexandre-allard committed:
salt: install yum-utils when configuring repos

we need to install yum-utils to provide repoquery
which is used in the macro pkg_installed() to be
able to lookup for packages and their dependencies
and enforce certain versions.

Refs: #2334

alexandre-allard committed:
scripts: update solutions configmap at archive import

Update automatically the metalk8s-solutions configmap
when importing or removing a Solution from the cluster,
so the configmap always reflects what solutions are
available.

Refs: #2344

TeddyAndrieux committed:
eve: Add TestRail description file and objects creation stage

Add a YAML file describing the TestRail objects that need to be created.
And run tool from test_tools repository, to create all TestRail objects
from this YAML description file.

Fixes: #2335

TeddyAndrieux committed:
eve: Set `metalk8s_short_version` property during eve build

In some case we may need to have the MetalK8s short version available to
use in some tools (like TestRail objects creation, ...).
Add a new property `metalk8s_short_version` equal to current short
version tested.

TeddyAndrieux committed:
eve: Make `TEST_NAME` and `CLASS NAME` more user friendly

Remove `-` in `TEST_NAME` and `CLASS NAME` to have more user friendly
naming and better naming in TestRail.

TeddyAndrieux committed:
eve: Add basic kube pod and basic docker centos7 image

Add a simple Dockerfile for a basic docker image from centos:7 able to
run as eve worker (with an eve user, eve workspace directory, ...).

TeddyAndrieux committed:
eve: Add TestRail upload results step

Add a step to upload TestRail results using JUnit file generated at
build time. This upload of JUnit file is done by a tool from test_tools
repository.
NOTE: Ideally this should be directly part of `eve`

Fixes: #2336

TeddyAndrieux committed:
eve: Make `TEST_NAME` and `CLASS NAME` more user friendly …

Remove `-` in `TEST_NAME` and `CLASS_NAME` to have more user friendly
naming and better naming in TestRail.

TeddyAndrieux committed:
eve: Add TestRail object for Patch upgrade and downgrade

Add information for Patch upgrade and downgrade to TestRail YAML
description file.

Sees: #2335

TeddyAndrieux committed:
eve: Add TestRail object for Example solution build

Add information for Example solution build to TestRail YAML
description file.

Sees: #2335

wabernat committed:
documentation: clarified bootstrap instructions further.

wabernat committed:
documentation: clarified bootstrap instructions further.

Ebaneck committed:
kubernetes: Bump kubernetes version

Closes: #2328
Closes: #2327

gdemonet committed:
deps: Regenerate compiled requirements

alexandre-allard committed:
salt: add labels to solution operator and ui objects

add all kubernetes default labels to the solution
operator and UI objects

Refs: #2366

alexandre-allard committed:
salt: fix deployment of Solutions UI ingress

Refs: #2365

alexandre-allard committed:
tests: adapt example solution operator test label

change from `name` to `app` to reflect the labels
changes

Refs: #2366

alexandre-allard committed:
salt: add labels by default on every object

add `heritage` and `app.kubernetes.io/managed-by`
labels with `salt` as the value on each and every
objects we deploy using our Kubernetes module

TeddyAndrieux committed:
eve: Add `message` support for JUnit generation

For each eve stage we generate a JUnit with one TestCase, we may want to
add a message in this TestCase, it's now doable with `MESSAGE`
environement variable

TeddyAndrieux committed:
eve: Add `text` support for JUnit generation

For each eve stage we generate a JUnit with one TestCase, we may want to
add a text in this TestCase, it's now doable with `TEXT` environment
variable

TeddyAndrieux committed:
eve: Add `text` in JUnit containing artifact Url

Currently we use JUnit file to upload test result to TestRail but we
didn't have any information to investigate in case of test fail. Add
artifact Url in the JUnit to upload this Url to TestRail, this is a
first step waiting for #2371

Sees: #2371

gdemonet committed:
deps: Pin pytest version

It seems the handling of nested fixtures with pytest-bdd doesn't work
well with pytest 5.

TeddyAndrieux committed:
eve: Add branch and commit revision to `text` in JUnit

Currently we use JUnit file to upload test result to TestRail but we
only have artifacts URL to investigate in case of test fail. Add
branch name and commit revision in the JUnit to upload this to
TestRail, this is a second step waiting for #2371

Sees: #2371

alexandre-allard committed:
salt: add an orchestration to import solutions

this orchestration allow to do all the import
related stuff in one place (except the mount of
the solution into the salt-master container,
because we can't restart the master during
an orchestration).

Refs: #2344

alexandre-allard committed:
scripts: use the new import solution orchestration

Refs: #2344

TeddyAndrieux committed:
docs: Rename "Development Best Practices" to "Development"

Rename "Development Best Practices" to "Development" as this section
could contain documentation that is not only Best Practices but real
development process

TeddyAndrieux committed:
docs: Add developer docs about CI testing

Add a developer documentation describing how to add a new test stage in
the CI to generate a proper JUnit result file and create all related
TestRail objects and upload result to TestRail if needed.

Fixes: #2346

wabernatScality commented:
.

wabernatScality commented:
@bert-e help

bert-e commented:
# Help page



The following options and commands are available at this time.

## Options ##


name   | description  | privileged   | authored
------ | ------------ | ------------ |----------
:arrow_right: **create_pull_requests** | Allow the creation of integration pull requests. |  | 
:arrow_right: **bypass_author_approval** | Bypass the pull request author's approval |  :star:  | 
:arrow_right: **bypass_commit_size** | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
:arrow_right: **unanimity** | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
:arrow_right: **bypass_incompatible_branch** | Bypass the check on the source branch prefix |  :star:  | 
:arrow_right: **bypass_build_status** | Bypass the build and test status |  :star:  | 
:arrow_right: **approve** | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
:arrow_right: **after_pull_request** | Wait for the given pull request id to be merged before continuing with the current one. |  | 
:arrow_right: **wait** | Instruct Bert-E not to run until further notice. |  | 
:arrow_right: **bypass_jira_check** | Bypass the Jira issue check |  :star:  | 
:arrow_right: **no_octopus** | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
:arrow_right: **bypass_leader_approval** | Bypass the pull request leaders' approval |  :star:  | 
:arrow_right: **bypass_peer_approval** | Bypass the pull request peers' approval |  :star:  | 



## Commands ##


name   | description  | privileged
------ | ------------ | ------------
:arrow_right: **reset** | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
:arrow_right: **help** | Print Bert-E's manual in the pull request. | 
:arrow_right: **build** | Re-start a fresh build ```TBA``` | 
:arrow_right: **force_reset** | Delete integration branches & pull requests, and restart merge process from the beginning. | 
:arrow_right: **retry** | Re-start a fresh build ```TBA``` | 
:arrow_right: **status** | Print Bert-E's current status in the pull request ```TBA``` | 
:arrow_right: **clear** | Remove all comments from Bert-E from the history ```TBA``` |

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

wabernat commented:
I do not know how to move the testing forward on this PR. I'm seeing that some checks haven't completed yet, but don't know how to find, resolve, or bypass the test issue.

TeddyAndrieux committed:
eve: Add build url to `text` in JUnit

Currently we use JUnit file to upload test result to TestRail but we
only have artifacts URL, branch name and commit revision to investigate in
case of test fail. Add build url in the JUnit to upload this to TestRail

Fixes: #2371

alexandre-allard committed:
build: add import-components SLS to build chain

Refs: #2344

wabernat commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Branches have diverged



This pull request's source branch `documentation/edit_bootstrap` has diverged from
`development/2.4` by more than 50 commits.

To avoid any integration risks, please re-synchronize them using one of the
following solutions:

* Merge `origin/development/2.4` into `documentation/edit_bootstrap`
* Rebase `documentation/edit_bootstrap` onto `origin/development/2.4`

Note: If you choose to rebase, you may have to ask me to rebuild
integration branches using the `reset` command.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `documentation/edit_bootstrap` into
`development/2.4`
* [w/2.5/documentation/edit_bootstrap](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/documentation/edit_bootstrap) will be merged into `development/2.5` 
* [w/2.6/documentation/edit_bootstrap](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/documentation/edit_bootstrap) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote

wabernat committed:
Fixed linter errors

wabernat committed:
Fixed linter errors

wabernat committed:
Merge branch 'documentation/edit_bootstrap' of http://github.com/Scality/metalk8s into documentation/edit_bootstrap

wabernat commented:
Rebased on to development/2.4 in hopes of beating build issue.

wabernat committed:
Merge branch 'development/2.4' into documentation/edit_bootstrap

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.5/documentation/edit_bootstrap` with contents from `documentation/edit_bootstrap`
and `development/2.5`.

Please resolve the conflict on **the integration branch** (`w/2.5/documentation/edit_bootstrap`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.5/documentation/edit_bootstrap
 $ git pull  # or "git reset --hard origin/w/2.5/documentation/edit_bootstrap"
 $ git merge origin/development/2.5
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/documentation/edit_bootstrap
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.5/documentation/edit_bootstrap
```

bert-e commented:
# Conflict




A conflict has been raised during the update of
integration branch `w/2.6/documentation/edit_bootstrap` with contents from `w/2.5/documentation/edit_bootstrap`
and `development/2.6`.

Please resolve the conflict on **the integration branch** (`w/2.6/documentation/edit_bootstrap`).


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout w/2.6/documentation/edit_bootstrap
 $ git pull  # or "git reset --hard origin/w/2.6/documentation/edit_bootstrap"
 $ git merge origin/development/2.6
 $ # <intense conflict resolution>
 $ git commit
 $ git merge origin/w/2.5/documentation/edit_bootstrap
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/2.6/documentation/edit_bootstrap
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote

miniscruff reviewed (approved):
:+1:

wabernat commented:
@bert-e approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nootal

* @alexandre-allard-scality

* @lucieleonard

* @sayf-eddine-scality

* @slaperche-scality

* @nicolast

* @gdemonet

* @carlito-scality

* @thomasdanan

* @jbertran

* @teddyandrieux

* @cmonfort

* @monpote










*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality










*The following options are set:* **approve**
